src/main/java/au/org/emii/ncdfgenerator/NcdfEncoderBuilder.java

```diff
@@ -0,0 +1,1476 @@
+				}
+				else if( clazz.equals( Long.class )) {
+					instance_id = (long)(Long)o;
+				} else {
+					throw new NcdfGeneratorException( "Can't convert intance_id type to integer" );
+				}
+
+				System.out.println( "whoot get(), instance_id is " + instance_id );
+
+				String selection = translate.process( selection_expr); // we ought to be caching the specific query ???
+
+				populateValues( definition.dimensions, definition.encoders,
+					"SELECT * FROM (" + definition.virtualInstanceTable + ") as instance where instance.id = " + Long.toString( instance_id) );
+
+
+				// is the order clause in sql part of projection or selection ?
+
+				// eg. concat "," $ map (\x -> x.getName) dimensions.values ...
+				String dimensionVar = "";
+				for( IDimension dimension : definition.dimensions.values() )
+				{
+					if( ! dimensionVar.equals("")){
+						dimensionVar += ",";
+					}
+					dimensionVar += "\"" + dimension.getName() + "\"" ;
+				}
+
+				populateValues( definition.dimensions, definition.encoders,
+					"SELECT * FROM (" + definition.virtualDataTable + ") as data where " + selection +  " and data.instance_id = " + Long.toString( instance_id) + " order by " + dimensionVar  );
+
+				NetcdfFileWriteable writer = createWritable.create();
+
+
+				for ( IDimension dimension: definition.dimensions.values()) {
+					dimension.define(writer);
+				}
+
+				for ( IVariableEncoder encoder: definition.encoders.values()) {
+					encoder.define( writer );
+				}
+				// finish netcdf definition
+				writer.create();
+
+				for ( IVariableEncoder encoder: definition.encoders.values()) {
+					// maybe change name writeValues
+					encoder.finish( writer );
+				}
+				// write the file
+				writer.close();
+
+				// TODO we should be returning a filestream here...
+				// the caller doesn't care that it's a netcdf
+				return writer;
+			}
+			else {
+				// no more netcdfs
+				conn.close();
+				return null;
+			}
+		} catch ( Exception e ) {
+			System.out.println( "Opps " + e.getMessage() );
+			conn.close();
+			return null;
+		}
+	}
+}
+
+
+public class NcdfEncoderBuilder
+{
+	// responsible for assembling the NcdfEncoder
+
+	public NcdfEncoderBuilder()
+	{
+		this.parser = new ExprParser();
+		this.translate = new  PGDialectTranslate();
+		this.createWritable = new CreateWritable( "testWrite.nc");
+	}
+
+	final IExprParser parser;
+	final IDialectTranslate translate;
+	final ICreateWritable createWritable;
+
+
+	public NcdfEncoder create ( InputStream config, String filterExpr, Connection conn ) throws Exception
+	{
+		// not sure if the expression parsing shouldn't go in here?
+		// not sure if definition decoding should be done here...
+
+		NcdfDefinition definition = null;
+		try {
+			Document document = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(config);
+			Node node =	document.getFirstChild();
+			definition = new NcdfDefinitionXMLParser().parseDefinition( node );
+
+		} finally {
+			config.close();
+		}
```

Comment:
I got very close to trying Jackson, since I would have preferred json, but elected to hand-code just because it would be faster. Agree this decision should be revisited.